### PR TITLE
Liveshare client

### PR DESCRIPTION
**Pull request by dibarbet**
Copying client side over to roslyn.

Note that there are still issues with goto def, find usages so they have been commented out.  Additionally, there is a bug with the liveshare public API, so I have not been able to test each individual handler.

Probably easiest to review commit by commit.

### CODE REVIEW

File: src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs [hunk 1]
```diff
@@ -57,6 +57,20 @@ internal static class ProtocolConversions
             { WellKnownTags.NuGet, LSP.CompletionItemKind.Text }
         };
 
+        public static LSP.TextDocumentPositionParams PositionToTextDocumentPositionParams(int position, SourceText text, Document document)
```
Comment (heejaechang) [inline src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs]: any reason these not being extension method?

File: src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj [hunk 2]
```diff
@@ -91,6 +91,7 @@
          and this should be removed. -->
     <InternalsVisibleTo Include="RoslynETAHost" WorkItem="https://github.com/dotnet/roslyn/issues/35072" />
     <InternalsVisibleTo Include="RoslynTaoActions" WorkItem="https://github.com/dotnet/roslyn/issues/35072" />
+    <InternalsVisibleTo Include="Microsoft.CodeAnalysis.ExternalAccess.LiveShare" />
```
Comment (heejaechang) [inline src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj]: now with this, we should remove any other LiveShare IVT.

Comment (heejaechang) [reply src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj]: but, let's do that in different PR with review from live share team.

File: src/Tools/ExternalAccess/LiveShare/Navigation/RoslynNavigationBarItemService.Exports.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        [ImportingConstructor]
+        protected VBLspNavigationBarItemService(RoslynLSPClientServiceFactory roslynLSPClientServiceFactory)
+            : base(roslynLSPClientServiceFactory)
+        {
+        }
+    }
+
+    [ExportLanguageService(typeof(INavigationBarItemService), StringConstants.TypeScriptLanguageName, WorkspaceKind.AnyCodeRoslynWorkspace), Shared]
+    internal class TypeScriptLspNavigationBarItemService : RoslynNavigationBarItemService
```
Comment (heejaechang) [inline src/Tools/ExternalAccess/LiveShare/Navigation/RoslynNavigationBarItemService.Exports.cs]: so is this only one left for typescript? when anycode roslyn workspace used except here?

File: src/Tools/ExternalAccess/LiveShare/RoslynLSPClientService.cs [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    internal class RoslynLSPClientServiceFactory : ICollaborationServiceFactory
+    {
+        private const string RoslynProviderName = "Roslyn";
+        private const string AnyProviderName = "any";
+
+        public ILanguageServerClient ActiveLanguageServerClient { get; private set; }
+
+        public Task<ICollaborationService> CreateServiceAsync(CollaborationSession collaborationSession, CancellationToken cancellationToken)
+        {
+            var languageServerGuestService = (ILanguageServerGuestService)collaborationSession.GetService(typeof(ILanguageServerGuestService));
```
Comment (heejaechang) [inline src/Tools/ExternalAccess/LiveShare/RoslynLSPClientService.cs]: what should we do here if we want to separate out type script here.

Comment (heejaechang) [reply src/Tools/ExternalAccess/LiveShare/RoslynLSPClientService.cs]: tagging @daytonellwanger

File: src/Tools/ExternalAccess/LiveShare/Navigation/RoslynNavigationBarItemService.Exports.cs (hunk 3 above)
Comment (dibarbet) [reply src/Tools/ExternalAccess/LiveShare/Navigation/RoslynNavigationBarItemService.Exports.cs]: This one didn't have a pragma to remove it

Not sure why - maybe @daytonellwanger might know?

Comment (daytonellwanger) [reply src/Tools/ExternalAccess/LiveShare/Navigation/RoslynNavigationBarItemService.Exports.cs]: We still need Roslyn for the NavBar. LSP doesn't yet support it. This is the last feature that we haven't moved for TS.

File: src/Tools/ExternalAccess/LiveShare/RoslynLSPClientService.cs (hunk 4 above)
Comment (daytonellwanger) [reply src/Tools/ExternalAccess/LiveShare/RoslynLSPClientService.cs]: TypeScript is already separated out here. The only TypeScript-specific code is lines 51-78. Once you drop TypeScript support, you'd just delete those lines.

**Comment by dibarbet** [comment]
done in different PR

**Pull request closed without merging**